cmd/drone-server/drone.go

```diff
@@ -236,6 +244,14 @@ func main() {
 		auth.POST("", server.GetLogin)
 	}
 
+	redirects := r.Group("/redirect")
+	{
+		redirects.Use(server.SetDatastore(store))
+		redirects.Use(server.SetRepo())
+
+		redirects.GET("/:owner/:name/commit/:sha", server.RedirectSha)
```

Comment:
I guess I'm still a bit confused by all of this ... it seems we need to add a ton of GitLab-specific code to Drone? I guess I don't understand what all of this code does since I don't use GitLab. Can you please further explain in the comments why all of this is required? Thanks!